pycqed/analysis/PSD/standard_arches_psd.py

```diff
@@ -0,0 +1,333 @@
+    Gamma_echo = 1.0/Techo[~exclusion_mask]
+
+    Gamma_phi_ramsey = Gamma_ramsey - Gamma_1/2.0
+    Gamma_phi_echo = Gamma_echo - Gamma_1/2.0
+
+    plot_coherence_times(flux, freq, sensitivity,
+                         T1, Tramsey, Techo, path)
+    plot_ratios(flux, freq, sensitivity,
+                Gamma_phi_ramsey, Gamma_phi_echo, path)
+
+    fit_res_gammas = fit_gammas(sensitivity, Gamma_phi_ramsey, Gamma_phi_echo)
+
+    intercept = fit_res_gammas.params['intercept'].value
+    slope_ramsey = fit_res_gammas.params['slope_ramsey'].value
+    slope_echo = fit_res_gammas.params['slope_echo'].value
+
+    plot_gamma_fit(sensitivity, Gamma_phi_ramsey, Gamma_phi_echo,
+                   slope_ramsey, slope_echo, intercept, path)
+
+    # after fitting gammas
+    # from flux noise
+    A = (slope_echo/(2*np.pi))/np.sqrt(np.log(2))
+    print('Amplitude PSD = (%s u\Phi_0)^2' % (A/1e-6))
+
+    # from white noise
+    # using Eq 5 in Nature com. 7,12964 (The flux qubit reviseited to enhanbce
+    # coherence and reporducibility)
+    if not ((freq_resonator is None) and (Qc is None) and (chi_shift is None)):
+        n_avg = calculate_n_avg(freq_resonator, Qc, chi_shift, intercept)
+        print('Estimated residual photon number: %s' % n_avg)
+    else:
+        n_avg = np.nan
+
+    return (A/1e-6), n_avg
+
+def calculate_n_avg(freq_resonator, Qc, chi_shift, intercept):
+    """
+    Returns the avg photon of white noise,assuming photon shot noise from the RO hanger.
+    """
+    k_r = 2*np.pi*freq_resonator/Qc
+    eta = k_r**2/(k_r**2 + 4*chi_shift**2)
+    n_avg = intercept*k_r/(4*chi_shift**2*eta)
+    return n_avg
+
+
+def prepare_input_table(dac, frequency, T1, T2_star, T2_echo,
+                        T1_mask=None, T2_star_mask=None, T2_echo_mask=None):
+    """
+    Returns a table ready for PSD_Analysis input
+    If sizes are different, it adds nans on the end.
+    """
+    assert(len(dac) == len(frequency))
+    assert(len(dac) >= len(T1))
+    assert(len(dac) >= len(T2_star))
+    assert(len(dac) >= len(T2_echo))
+
+    if T1_mask is None:
+        T1_mask = np.zeros(len(T1), dtype=np.bool)
+    if T2_star_mask is None:
+        T2_star_mask = np.zeros(len(T2_star), dtype=np.bool)
+    if T2_echo_mask is None:
+        T2_echo_mask = np.zeros(len(T2_echo), dtype=np.bool)
+
+    assert(len(T1) == len(T1_mask))
+    assert(len(T2_star) == len(T2_star_mask))
+    assert(len(T2_echo) == len(T2_echo_mask))
+
+    table = np.ones((6, len(dac)))
+    table = table * np.nan
+    table[0, :] = dac
+    table[1, :len(frequency)] = frequency
+    table[2, :len(T1)] = T1
+    table[3, :len(T2_star)] = T2_star
+    table[4, :len(T2_echo)] = T2_echo
+    table[5, :len(T1_mask)] = np.logical_or(np.logical_or(T1_mask,
+                                                          T2_star_mask),
+                                            T2_echo_mask)
+
+    return table
+
+
+def arch(dac, Ec, Ej, offset, dac0):
```

Comment:
Should use the arch as defined in the fit_mods, that should be updated to be consistent with this as per #143 (N.B. this version is correct).